services/src/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/multitable/MultitableFlowsSpec.groovy

```diff
@@ -0,0 +1,1203 @@
+            it.destination.vlanId = flow.destination.vlanId - 1
+        }
+        flowHelperV2.addFlow(flow2)
+
+        then: "Both existing flows are valid"
+        [flow, flow2].each {
+            northbound.validateFlow(it.flowId).each { assert it.asExpected }
+        }
+
+        and: "Involved switches pass switch validation"
+        involvedSwitches.each {
+            with(northbound.validateSwitch(it.dpId)) { validation ->
+                validation.verifyRuleSectionsAreEmpty(["missing", "excess"])
+                validation.verifyMeterSectionsAreEmpty(["missing", "excess", "misconfigured"])
+            }
+        }
+
+        and: "Both flows are pingable"
+        [flow, flow2].each {
+            verifyAll(northbound.pingFlow(it.flowId, new PingInput())) {
+                it.forward.pingSuccess
+                it.reverse.pingSuccess
+            }
+        }
+
+        and: "Both flows allow traffic"
+        def examFlow2 = new FlowTrafficExamBuilder(topology, traffExam)
+                .buildBidirectionalExam(flowHelperV2.toV1(flow2), 1000, 5)
+        withPool {
+            [examFlow1.forward, examFlow1.reverse, examFlow2.forward, examFlow2.reverse].eachParallel { direction ->
+                def resources = traffExam.startExam(direction)
+                direction.setResources(resources)
+                assert traffExam.waitExam(direction).hasTraffic()
+            }
+        }
+
+        when: "Delete flows"
+        def flow1InfoFromDb = database.getFlow(flow.flowId)
+        def flow2InfoFromDb = database.getFlow(flow2.flowId)
+        def flowsCookies = [flow1InfoFromDb.forwardPath.cookie.value, flow1InfoFromDb.reversePath.cookie.value,
+                            flow2InfoFromDb.forwardPath.cookie.value, flow2InfoFromDb.reversePath.cookie.value]
+        [involvedSwitches[0], involvedSwitches[3]].each { sw ->
+            northbound.getSwitchRules(sw.dpId).flowEntries.each { rule ->
+                Cookie.isIngressRulePassThrough(rule.cookie) && flowsCookies << rule.cookie
+            }
+        }
+        [flow, flow2].each { flowHelper.deleteFlow(it.flowId) }
+
+        then: "Flow rules are deleted from switches"
+        involvedSwitches.each { sw ->
+            with(northbound.getSwitchRules(sw.dpId).flowEntries) { rules ->
+                rules.findAll { it.cookie in flowsCookies }.empty
+            }
+        }
+
+        cleanup: "Revert system to origin state"
+        involvedSwitches.each { sw ->
+            //system leaves excess rules after removing multi table flow from a switch with single table mode
+            northbound.synchronizeSwitch(sw.dpId, true)
+            northbound.updateSwitchProperties(sw.dpId, northbound.getSwitchProperties(sw.dpId)
+                    .tap { it.multiTable = initSwProps[sw.dpId].multiTable })
+        }
+        northbound.deleteLinkProps(northbound.getAllLinkProps())
+        checkDefaultRulesOnSwitches(involvedSwitches)
```

Comment:
you sure that this should be a part of cleanup? At least right now this looks strange for me, need to hear your arguments